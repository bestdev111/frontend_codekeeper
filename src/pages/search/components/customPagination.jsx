import React from "react";
import { Box, TablePagination } from "@mui/material";

export const CustomPagination = ({
  total_counts,
  pageInfo,
  handleChangePage,
  handleChangeRowsPerPage,
}) => {
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
        textAlign: "center",
        gap: 2,
        "& .MuiTablePagination-root": {
          display: "flex",
          justifyContent: "center",
        },
        "@media (max-width: 445px)": {
          "& .MuiTablePagination-selectLabel": {
            display: "none",
          },
        },
      }}
    >
      <TablePagination
        component="div"
        count={total_counts}
        page={pageInfo.pageNum - 1}
        onPageChange={handleChangePage}
        rowsPerPage={pageInfo.pageSize}
        onRowsPerPageChange={handleChangeRowsPerPage}
      />
    </Box>
  );
};
