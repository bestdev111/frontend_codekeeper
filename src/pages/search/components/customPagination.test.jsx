import React from "react";
import { render } from "@testing-library/react";
import { CustomPagination } from "./customPagination";

jest.mock("@mui/material", () => ({
  Box: () => <div data-testid="mock-box" />,
  TablePagination: () => <div data-testid="mock-table-pagination" />,
}));

describe("<CustomPagination>", () => {
  it("should render component", () => {
    const pageInfo = { pageNum: 1, pageSize: 10 };
    const { container } = render(<CustomPagination pageInfo={pageInfo} />);
    expect(container).toMatchSnapshot();
  });
});
