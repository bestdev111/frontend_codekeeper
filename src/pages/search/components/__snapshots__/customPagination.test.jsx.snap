// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomPagination> should render component 1`] = `
<div>
  <div
    data-testid="mock-box"
  />
</div>
`;
